### PR TITLE
Removed support for EOL-versions of PHP and Symfony

### PR DESCRIPTION
**Pull request by stephanvierkant**
Other versions (<7.4) are now End of Life. 

MCrypt is deprecated since PHP 7.1 and removed in 7.2, so I removed support for it as well.

Closes #122

### CODE REVIEW

**Comment by gimler** [comment]
@stephanvierkant what about the other `symfony/*` dependencies? In `composer.json` for example `"symfony/framework-bundle": "~2.3|~v3.0|~4.0|~5.0|~6.0". i think we can also drop some of the old versions

**Comment by franmomu** [comment]
> @stephanvierkant what about the other `symfony/*` dependencies? In `composer.json` for example `"symfony/framework-bundle": "~2.3|~v3.0|~4.0|~5.0|~6.0". i think we can also drop some of the old versions

We can do that in another PR as it will require some other clean ups.

@Seldaek are you fine releasing this change (when it's ready) in `v2.11`?

File: Encrypter.php [hunk 1]
```diff
@@ -1,79 +0,0 @@
...
- * For the full copyright and license information, please view the LICENSE
- * file that was distributed with this source code.
- */
-
-namespace Nelmio\SecurityBundle;
-
-/**
- * @deprecated EncryptedCookieListener is now deprecated due to high coupling with the deprecated mcrypt extension
- */
-class Encrypter
```
Comment (franmomu) [inline Encrypter.php:17]: It's using the deprecated `mcrypt` extension, but someone could have extended it, if we're releasing this as `v2.11` I think we should keep them.

File: composer.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
         }
     ],
     "require": {
-        "php": ">5.4",
+        "php": ">7.4",
         "paragonie/random_compat": "~1.0|~2.0|~9.99.99",
```
Comment (franmomu) [inline composer.json:19]: I think we can also remove this package

File: composer.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
         }
     ],
     "require": {
-        "php": ">5.4",
+        "php": ">7.4",
```
Comment (franmomu) [inline composer.json:18]:
```suggestion
        "php": "^7.4|^8.0",
```

**Comment by Seldaek** [comment]
IMO either wait, clean up all the things and release a 3.0 breaking what needs to be, or release a minimal sf6 support minor version which does not break anything beyond perhaps dropping very old symfony /php

File: Encrypter.php (hunk 1 above)
Comment (stephanvierkant) [reply Encrypter.php:17]: This class has been marked deprecated in 5,5 years ago: [8b3e412be202ec0370ade05059d4515a2d316e84](https://github.com/nelmio/NelmioSecurityBundle/commit/8b3e412be202ec0370ade05059d4515a2d316e84). I think we can safely remove it?

Comment (Seldaek) [reply Encrypter.php:17]: It's fine to remove, but it means the next release should become 3.0, and if we do that we may take the chance to remove/cleanup other stuff if needed.

**Comment by franmomu** [comment]
I agree, let's release `2.11.0` with support of Symfony 6 (which was released some time ago) and then we start cleaning things up for a new major version with support of PHP >= 7.4.

**Comment by stephanvierkant** [comment]
@franmomu I agree! Let's release a new minor version. I'll continue with this PR for 3.x

**Comment by Seldaek** [comment]
👍🏻 thanks everyone for pushing forward here.

**Comment by franmomu** [comment]
I've rebased to resolve the conflicts and remove support of Twig 1 (it was the one making tests fail).

File: .github/workflows/continuous-integration.yml [hunk 4]
```diff
@@ -15,27 +15,22 @@ jobs:
...
-          - php-version: 5.5
+          - php-version: 7.4
             dependencies: lowest
-            symfony-version: 2.8
-          - php-version: 5.6
+            symfony-version: 4.4
+          - php-version: 7.4
             dependencies: highest
-            symfony-version: 2.8
+            symfony-version: 4.4
```
[the remark this answers is not in the document]
Comment (franmomu) [reply .github/workflows/continuous-integration.yml:33]: We have:

```
Continuous Integration / PHP 7.4 + Symfony 4.4.x + highest (pull_request)
```

and 

```
Continuous Integration / PHP 7.4 + Symfony 4.4 + highest (pull_request)
```
so we can indeed remove the last job from `include`

File: composer.json [hunk 5]
```diff
@@ -15,19 +15,18 @@
...
-        "symfony/security-core": "~2.3|~3.0|~4.0|~5.0|~6.0",
-        "symfony/security-http": "~2.3|~3.0|~4.0|~5.0|~6.0",
-        "symfony/security-csrf": "~2.3|~3.0|~4.0|~5.0|~6.0",
-        "symfony/yaml": "~2.3|~3.0|~4.0|~5.0|~6.0",
+        "php": "^7.4|^8.0",
+        "symfony/framework-bundle": "~4.4|~5.0|~6.0",
+        "symfony/security-core": "~4.4|~5.0|~6.0",
+        "symfony/security-http": "~4.4|~5.0|~6.0",
+        "symfony/security-csrf": "~4.4|~5.0|~6.0",
+        "symfony/yaml": "~4.4|~5.0|~6.0",
```
Comment (franmomu) [inline composer.json:23]:
```suggestion
        "symfony/framework-bundle": "~4.4|~5.0|~6.0",
        "symfony/security-core": "~4.4|~5.4|~6.0",
        "symfony/security-http": "~4.4|~5.4|~6.0",
        "symfony/security-csrf": "~4.4|~5.4|~6.0",
        "symfony/yaml": "~4.4|~5.4|~6.0",
```
Let's only allow supported versions (right now 5.3 is supported, but by the time we release a new major version it won't)

File: composer.json [hunk 6]
```diff
@@ -15,19 +15,18 @@
...
         "ua-parser/uap-php": "^3.4.4"
     },
     "require-dev": {
         "doctrine/cache": "^1.0",
         "psr/cache": "^1.0",
-        "twig/twig": "^1.38|^2.10|^3.0",
+        "twig/twig": "^2.10|^3.0",
         "symfony/phpunit-bridge": "^5.0.5|^6.0"
```
Comment (franmomu) [inline composer.json:30]:
```suggestion
        "symfony/phpunit-bridge": "^6.0"
```

**Review by franmomu — APPROVED** [review]
I think this is enough to start, there are still some BC layers with Symfony to be removed (at least in `EventListener`), but that can be done in another PR and merging this allow us to work on type declarations, coverage, etc more easily.

**Comment by franmomu** [comment]
thanks @stephanvierkant!

**Pull request merged**